source/nordic_sdk/components/ble/common/ble_srv_common.c

```diff
@@ -59,3 +59,99 @@ void ble_srv_ascii_to_utf8(ble_srv_utf8_str_t * p_utf8, char * p_ascii)
     p_utf8->length = (uint16_t)strlen(p_ascii);
     p_utf8->p_str  = (uint8_t *)p_ascii;
 }
+
+
+/**@brief Function for setting security requirements of a characteristic.
+ *
+ * @param[in]  level required security level.
+ *
+ * @return     encoded security level and security mode.
+ */
+static inline ble_gap_conn_sec_mode_t set_security_req(security_req_t level)
+{
+    ble_gap_conn_sec_mode_t perm;
+
+    BLE_GAP_CONN_SEC_MODE_SET_NO_ACCESS(&perm);
+    switch (level)
+    {
+        case SEC_NO_ACCESS:
+            BLE_GAP_CONN_SEC_MODE_SET_NO_ACCESS(&perm);
+        break;
+        case SEC_OPEN:
+            BLE_GAP_CONN_SEC_MODE_SET_OPEN(&perm);
+        break;
+        case SEC_JUST_WORKS:
+            BLE_GAP_CONN_SEC_MODE_SET_ENC_NO_MITM(&perm);
+        break;
+        case SEC_MITM:
+            BLE_GAP_CONN_SEC_MODE_SET_ENC_WITH_MITM(&perm);
+        break;
+        case SEC_SIGNED:
+            BLE_GAP_CONN_SEC_MODE_SET_SIGNED_NO_MITM(&perm);
+        break;
+        case SEC_SIGNED_MITM:
+            BLE_GAP_CONN_SEC_MODE_SET_SIGNED_WITH_MITM(&perm);
+        break;
+    }
+    return(perm);
+}
+
+
+uint32_t characteristic_add(uint16_t                   service_handle,

```

Comment:
is this a new helper functionality offered by the SDK?  can we make use of this in ble-nrf51822? i.e. in `nRF51822/source/btle/custom/custom_helper.cpp`? do the release notes mention this? We should look into this more carefully.